### PR TITLE
refactor(core): Reduce code duplication in DB config (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
Our DB config code has picked up a lot of dead and duplicate code over the last year. This PR simplifies how DB config is generated, and also updates tests to use an almost identical DB config as the application itself. 

## Review / Merge checklist
- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/cli/src/databases/config.ts [hunk 1]
```diff
@@ -51,28 +47,63 @@ export const getOptionOverrides = (dbType: 'postgresdb' | 'mysqldb') => ({
...
+		};
+	}
+
+	return {
+		type: 'postgres',
+		...getCommonOptions(),
+		...getOptionOverrides('postgresdb'),
+		schema: config.getEnv('database.postgresdb.schema'),
+		poolSize: config.getEnv('database.postgresdb.poolSize'),
+		migrations: postgresMigrations,
```
Comment (tomi) [inline packages/cli/src/databases/config.ts:83]: Should these be imported dynamically for each DB? Not sure if there is much perf impact tho

File: packages/cli/test/integration/shared/testDb.ts [hunk 2]
```diff
@@ -124,57 +82,16 @@ export async function truncate(names: Array<(typeof repositories)[number]>) {
...
-// ----------------------------------
-
 /**
- * Generate options for an in-memory sqlite database connection,
- * one per test suite run.
+ * Generate options for a bootstrap DB connection, to create and drop test databases.
  */
-const getSqliteOptions = ({ name }: { name: string }): ConnectionOptions => {
+export const getBootstrapDBOptions = (dbType: 'postgresdb' | 'mysqldb'): DataSourceOptions => {
+	const type = dbType === 'postgresdb' ? 'postgres' : 'mysql';
```
Comment (tomi) [inline packages/cli/test/integration/shared/testDb.ts:89]: We don't run these tests against SQLite? Or are they handled somehow differently?

**Review by tomi — APPROVED** [review]
Much cleaner now 👏

File: packages/cli/src/databases/config.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/src/databases/config.ts:83]: likely not going to make much difference in performance or memory usage.

**Comment by janober** [comment]
Got released with [`n8n@1.30.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.30.0)

**Pull request merged**